compiler/rustc_abi/src/lib.rs

```diff
@@ -967,6 +1006,28 @@ impl WrappingRange {
         }
     }
 
+    /// Returns `true` if `vmin..=vmax` is contained in `self`.
+    #[inline(always)]
+    pub fn contains_range(&self, vmin: u128, vmax: u128) -> bool {
+        assert!(vmin <= vmax);
+        if self.start <= self.end {
+            self.start <= vmin && vmax <= self.end
+        } else {
+            self.start <= vmin || vmax <= self.end || self.start == self.end + 1
+        }
+    }
+
+    /// Returns `true` if `vmin..=vmax` has an overlap with `self`.
+    #[inline(always)]
+    pub fn overlaps_range(&self, vmin: u128, vmax: u128) -> bool {
+        assert!(vmin <= vmax);
+        if self.start <= self.end {
+            self.start <= vmax && vmin <= self.end
+        } else {
+            self.start <= vmax || vmin <= self.end || self.start == self.end + 1
```

Comment:
Isn't the `self.start == self.end + 1` part her unnecessary? If `self.start <= vmax || vmin <= self.end` is false we have `end < vmin <= vmax < start`, to `end == start+1` is impossible.